### PR TITLE
feat: add ens name and avatar support

### PR DESCRIPTION
**Pull request by bigint**
image

### CODE REVIEW

**Comment by bigint** [comment]
cc @aminlatifi @MohammadPCh

**Comment by aminlatifi** [comment]
Thanks @bigint 
Sorry for the tardy response! I will review it

File: package.json [hunk 1]
```diff
@@ -4,6 +4,8 @@
   "private": true,
   "dependencies": {
     "@ant-design/icons": "^4.5.0",
+    "@davatar/react": "^1.8.1",
+    "@ethersproject/providers": "^5.5.2",
```
Comment (aminlatifi) [inline package.json:8]: We have web3 and our bundle is unfortunately already big! can we utilize the existing web3js instead?
I didn't find an alternative for `getDefaultProvider`! 

I want to suggest two options:
1. Using some reliable ens subgraph by which we will have less trouble I think
2. Switching to web3 and add an mainnet nodeurl to our setting if needed

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Again, thanks a lot @bigint
You are a hero who helps opensource non-profit projects

File: package.json (hunk 1 above)
Comment (bigint) [reply package.json:8]: @aminlatifi Sadly web3js dont have ens support :(

Comment (aminlatifi) [reply package.json:8]: > @aminlatifi Sadly web3js dont have ens support :(

Yeah, I understood the same!

Comment (bigint) [reply package.json:8]: @aminlatifi I think we can reduce it in further iterations, we really love to see this app on ens.domains homepage

once this is available in prod we will feature this app in https://ens.domains homepage and tweet about it

Comment (bigint) [reply package.json:8]: @aminlatifi can we move this forward or something to look later?

Comment (aminlatifi) [reply package.json:8]: > 

Yeah, I will merge it.

Comment (bigint) [reply package.json:8]: Love to feature u folks on ens.domains homepage

**Comment by aminlatifi** [comment]
> Again, thanks a lot @bigint You are a hero who helps opensource non-profit projects

@bigint but let me test first, unfortunately netlify cannot build and raises out of memory error!

**Comment by aminlatifi** [comment]
> > Again, thanks a lot @bigint You are a hero who helps opensource non-profit projects
> 
> @bigint but let me test first, unfortunately netlify cannot build and raises out of memory error!

Raised the issue here
https://answers.netlify.com/t/out-of-memory-on-build-react-app/50769

**Comment by bigint** [comment]
woah interesting error!

**Comment by bigint** [comment]
@aminlatifi seems like we got an update https://answers.netlify.com/t/out-of-memory-on-build-react-app/50769/2

**Comment by aminlatifi** [comment]
Yes, thanks for reminding. I am on vacation right now, then will try to spend some time on this.

**Comment by aminlatifi** [comment]
Thanks @bigint 
I could resolve the issue just today

**Pull request merged**